File: occasum-core/src/main/java/pw/stamina/occasum/node/instructions/PropertyPropertyNodeInstruction.java
package pw.stamina.occasum.node.instructions;

import pw.stamina.occasum.PropertyHandle;
import pw.stamina.occasum.node.PropertyNode;
import pw.stamina.occasum.properties.Property;

final class PropertyPropertyNodeInstruction implements PropertyNodeInstruction {
    private final PropertyHandle handle;
    private final Property property;

    PropertyPropertyNodeInstruction(PropertyHandle handle, Property property) {
        this.handle = handle;
        this.property = property;
    }

    @Override
    public void apply(PropertyNode node) {
        if (handle != null) {
            node.property(handle, property);
        } else {
            node.property(property);
        }
    }
}

File: occasum-core/src/main/kotlin/pw/stamina/occasum/node/instructions/PropertyPropertyNodeInstruction.kt
package pw.stamina.occasum.node.instructions

import pw.stamina.occasum.PropertyHandle
import pw.stamina.occasum.node.PropertyNode
import pw.stamina.occasum.properties.Property

internal class PropertyPropertyNodeInstruction(
        private val handle: PropertyHandle?,
        private val property: Property<*>
) : PropertyNodeInstruction {

    override fun apply(node: PropertyNode) {
        if (handle != null) {
            node.property(handle, property)
        } else {
            node.property(property)
        }
    }
}
